staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -288,6 +298,40 @@ func GetHistogramVecFromGatherer(gatherer metrics.Gatherer, metricName string, l
 	return vec, nil
 }
 
+type LabeledHistogram struct {
+	Labels    map[string]string
+	Histogram Histogram
+}
+
+// GetHistogramFromGatherer collects a metric from a gatherer implementing k8s.io/component-base/metrics.Gatherer interface.
+// If the metric is a HistogramVec, each metric from the HistogramVec will be collected with a label map one by one.
+// Used only for testing purposes where we need to gather metrics directly from a running binary (without metrics endpoint).
+func GetHistogramFromGatherer(gatherer metrics.Gatherer, metricName string, lvMap map[string]string) ([]LabeledHistogram, error) {
+	metricFamily, err := validateMetricFamily(gatherer, metricName)
+	if err != nil {
+		return nil, err
+	}
+
+	var histogramList []LabeledHistogram
+	for _, metric := range metricFamily.GetMetric() {
+		metricLabels := map[string]string{}
+		labelPairs := map[string]string{}
+		if LabelsMatch(metric, lvMap) {
+			for _, labelPair := range metric.Label {
+				metricLabels[labelPair.GetName()] = labelPair.GetValue()
+			}
+			for label := range lvMap {
+				labelPairs[label] = metricLabels[label]
+			}
+			histogramList = append(histogramList, LabeledHistogram{
```

Comment:
It looks like you are returning all the metrics (for all combinations of label values) each as a separate item.

Is there any aggregation happening? I don't see any in `collectHistogram` either. I think this method should be returning `[]HistogramVec`.

Can we add unit tests for this method?

Reply:
> for all combinations of label values

sort of, for the metrics defined as this,  ```"scheduler_framework_extension_point_duration_seconds": {"extension_point": "*"},``` we need to collect all of the metrics with the label "extension_point". `collectHistogram` func was added beside `collectHistogramVec` is intentionally for this purpose, shouldn't do any of aggregation, but collect all the metric with that label intead.

`collectHistogramVec` doesn't suit for the case of `extension_points`,  we use to be able to collect of the metrics related with this label, we'd want to collect each of them instead of aggregating them into one,

`collectHistogramVec` have is something like this, 
```[]HistogramVec = {"Histogram(score)",  "Histogram(filter)", etc}```

and calculate the Quantile based on the `[]HistogramVec`, this is meaning for the extension points iiuc.

Reply:
I have tested with this PR and the report has been attached in the commit description, you can take a look for how it looks like.

Reply:
+1 to what Aldo suggested, you cannot assume the matched (e.g., extension_point='*') metrics has **only one** {label/val} pair - it may have multiple {label/val} combinations like attempts, status, etc.